### PR TITLE
Added API for organisation (WIP)

### PR DESCRIPTION
**Pull request by Rupeshiya**
---
name: REST API FOR ORGANISATION
about: added API for organization
title: ''API FOR ORGS
labels: ''feature
assignees: ''@Rupeshiya

---

### **Problem**
Github Issue Number: #93 
Implementation of rest api for organisation_

### **Solution of problem**
added API for organization

### **Type of Change**
[ ] Bug fix
[&check;] New Feature
[ ] Development of UI/UX prototypes
[ ] Small refactor
[ ] Change in Documentation

### **Checklist**
[&check;] My code follows the same style as the codebase
[ ] My Code change requires a change in documentation
[ ] I have updated the Readme accordingly
[&check;] I made PR against **development branch**
[&check; ] I have run the test cases locally and it's passing.
[&check;] I have squashed my commits

### CODE REVIEW

**Comment by Rupeshiya** [comment]
@devesh-verma Why CI is not showing the test result correctly?
It's passing. Please check the Dashboard of CI
ci

**Comment by Rupeshiya** [comment]
@devesh-verma @vaibhavdaren Please review the PR!

File: app/models/Organisation.js [hunk 1]
```diff
@@ -87,29 +100,24 @@ const orgSchema = new Schema({
...
-    required: true,
-    validate (adminInfo) {
-      if (validator.isEmpty(adminInfo)) {
-        throw new Error('Admin info is required!')
-      }
-    }
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'User'
+  },
+  isArchived: {
```
Comment (devesh-verma) [inline app/models/Organisation.js:118]: why do we need this `isArchived`

File: app/controllers/organization.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
+const Organization = require('../models/Organisation')
+const handler = require('../utils/response-helper')
+const STATUS = require('../utils/status-codes')
+const helper = require('../utils/uploader')
```
Comment (devesh-verma) [inline app/controllers/organization.js:4]: the variables should have same style.

Comment (Rupeshiya) [reply app/controllers/organization.js:4]: @devesh-verma I usually do UPPERCASE for constants and lowercase for others.
So should I change that?

File: app/controllers/organization.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        .sort({ createdAt: -1 })
+        .exec()
+      if (orgsData.length === 0) {
+        return res.status(STATUS.OK).json({ msg: 'No organizations listed yet!' })
+      }
+      res.status(STATUS.OK).json({ organizations: orgsData })
+    } catch (error) {
+      handler.handleError(res, error)
+    }
+  },
```
Comment (devesh-verma) [inline app/controllers/organization.js:78]: we don't need this route at all, as donut is going to be hosted by the org itself

File: app/models/Organisation.js (hunk 1 above)
Comment (Rupeshiya) [reply app/models/Organisation.js:118]: @devesh-verma I have added that if an admin wants then they can either delete the organization completely or keep that in archived mode.
What say @devesh-verma ?

File: app/controllers/organization.js (hunk 2 above)
Comment (devesh-verma) [reply app/controllers/organization.js:4]: actually we are declaring everything as a constant which will make our lives miserable going forward.
As of now, handler and status always capital.
helper let it be like that.

File: app/models/Organisation.js [hunk 4]
```diff
@@ -60,7 +60,7 @@ const orgSchema = new Schema({
     }
   },
   contactInfo: {
-    emailId: {
+    communityEmail: {
```
Comment (devesh-verma) [inline app/models/Organisation.js:63]: this should be just email as it is already inside community we dont need to specify for what this email id.

File: app/models/Organisation.js [hunk 5]
```diff
@@ -72,6 +72,19 @@ const orgSchema = new Schema({
...
+      required: true,
+      validate (adminEmail) {
+        if (validator.isEmpty(adminEmail)) {
+          throw new Error('Admin emailId is required!')
+        }
+        if (!validator.isEmail(adminEmail)) {
+          throw new Error('Invalid emailId')
+        }
+      }
+    },
```
Comment (devesh-verma) [inline app/models/Organisation.js:87]: don't do this.

File: app/models/Organisation.js (hunk 1 above)
Comment (devesh-verma) [reply app/models/Organisation.js:118]: cool

File: app/models/Organisation.js [hunk 6]
```diff
@@ -47,7 +47,7 @@ const orgSchema = new Schema({
     }
   },
   logo: {
-    type: Buffer,
+    image: Buffer,
```
Comment (devesh-verma) [inline app/models/Organisation.js:50]: add one more field here imgUrl, look in other collections what we have named it and use the same name. this is for saving the aws link of the img

File: app/routes/organisation.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  auth,
+  OrgController.createOrganization
+)
+
+// GET ALL ORG
+router.get(
+  '/all',
+  auth,
+  OrgController.getAllOrg
+)
```
Comment (devesh-verma) [inline app/routes/organisation.js:20]: you can remove this

**Comment by Rupeshiya** [comment]
Done @devesh-verma !

**Pull request merged**